Description: Fix bug in handling filenames sharing a common prefix when renaming.
 This bug fix is imported from unreleased upstream and will be integrated in
 coming 0.54 release.  Closes: #654326.
 As stated by Kushal:
 .
    Here's a small patch that fixes debian bug 654326.  vidir modifies its
    in-memory copy of filenames in a way that can clobber items that have
    common prefix.  There's a one-liner testcase in the bug report.  This
    patch fixes it so regexp replacement is anchored at end of line or path
    component.
Author: Kushal Kumaran <kushal@locationd.net>
Bug-Debian: https://bugs.debian.org/654326
Origin: upstream, http://git.kitenet.net/?p=moreutils.git;a=commit;h=0aeceb285c9ce53a78c3b2fac8ff4b2015a0965d
Last-Update: 2014-11-10

--- moreutils-0.53.orig/vidir
+++ moreutils-0.53/vidir
@@ -188,7 +188,7 @@ while (<IN>) {
 			else {
 				if (-d $name) {
 					foreach (values %item) {
-						s/^\Q$src\E/$name/;
+						s,^\Q$src\E($|/),$name$1,;
 					}
 				}
 				if ($verbose) {
